config/header-snippets/types_legacy.h.in

```diff
@@ -85,9 +85,14 @@ typedef signed long long duk_int64_t;
     (defined(DUK_F_MINGW) || defined(DUK_F_MSVC))
 /* Both MinGW and MSVC have a 64-bit type. */
 #define DUK_F_HAVE_64BIT
+#if defined(DUK_F_MSVC)
+typedef unsigned __int64 duk_uint64_t;
+typedef signed __int64 duk_int64_t;
+#else
 typedef unsigned long duk_uint64_t;
 typedef signed long duk_int64_t;
 #endif
+#endif
```

Comment:
Because the MSVC and MinGW behavior no longer has anything in common, refactor to:

```c
#if !defined(DUK_F_HAVE_64BIT) && defined(DUK_F_MINGW)
#define DUK_F_HAVE_64BIT
typedef unsigned long duk_uint64_t;
typedef signed long duk_int64_t;
#endif
#if !defined(DUK_F_HAVE_64BIT) && defined(DUK_F_MSVC)
#define DUK_F_HAVE_64BIT
typedef unsigned __int64 duk_uint64_t;
typedef signed __int64 duk_int64_t;
#endif
```